spec.html

```diff
@@ -142,7 +142,20 @@ <h1>Updated Syntactic Grammar Productions</h1>
         CallExpression[?Yield, ?Await] `.` IdentifierName
         CallExpression[?Yield, ?Await] TemplateLiteral[?Yield, ?Await]
         <ins>CallExpression[?Yield, ?Await] `.` PrivateIdentifier</ins>
+
+      OptionalChain[Yield, Await] :
+        `?.` `[` Expression[+In, ?Yield, ?Await] `]`
+        `?.` IdentifierName
+        `?.` Arguments[?Yield, ?Await]
+        `?.` TemplateLiteral[?Yield, ?Await, +Tagged]
+        OptionalChain[?Yield, ?Await]  `[` Expression[+In, ?Yield, ?Await] `]`
+        OptionalChain[?Yield, ?Await] `.` IdentifierName
+        OptionalChain[?Yield, ?Await] Arguments[?Yield, ?Await]
+        OptionalChain[?Yield, ?Await] TemplateLiteral[?Yield, ?Await, +Tagged]
+        <ins>OptionalChain[?Yield, ?Await] `.` PrivateIdentifier</ins>
     </emu-grammar>
+
+    <emu-note>`?.` PrivateIdentifier is currently a unstated Syntax Error</emu-note>
```

Comment:
Another way to phrase this is,

> `?.` |PrivateIdentifier| is not part of the grammar for |OptionalChain|; therefore, expressions like `x?.#y` cause a SyntaxError when parsed as an |Expression|.

Depends how pedantic you and the editors want to be. But in particular, I don't think "currently" is needed--everything in the spec or proposal is "currently".

Reply:
is there any reason to not make `x?.#y` valid?

Reply:
Let's follow up with discussion in https://github.com/tc39/proposal-class-fields/issues/302.